storage_scrubber/src/checks.rs

```diff
@@ -84,7 +84,12 @@ pub(crate) fn branch_cleanup_and_check_errors(
                             .push(format!("index_part.json version: {}", index_part.version()))
                     }
 
-                    if &index_part.version() != IndexPart::KNOWN_VERSIONS.last().unwrap() {
+                    let newest_versions = IndexPart::KNOWN_VERSIONS
```

Comment:
I feel we should only retain one version here?

Reply:
During deploy we will inevitably have to deal with versions older than the latest one, and due to attaches of detached tenants, we'll encounter these also after a finished deploy of the newest pageserver everywhere, at least as long as we don't figure out a way to migrate the index part files in some other fashion.

Note that the only "migration" we would be doing in this instance is the increase of the version, as the `archived_at` field will be omitted anyways.

As it's nothing unexpected for the scrubber to find the version number before the latest one, nor is it anything bad, I don't think it should warn about it.

Note that this code is not about deleting old *generations* of index files, it's about warning index files with old version numbers. Which isn't as actionable as old generations.

Reply:
Why collect them to a vec though? Could just find from after `take(2)`.

Anyways, I'll need to bump the version soon after your PR, if mine ever gets reviewed :)

Reply:
> I'll need to bump the version soon after your PR

hmmmm interesting, if we manage to put both PRs into the same release we can maybe just bump it once?

collecting to a vec isn't strictly necessary and probably wasteful, you are right. I'll use `find`.

Reply:
using find now.